tools/src/misc/h5mkgrp.c

```diff
@@ -252,18 +252,24 @@ parse_command_line(int argc, const char *const *argv, mkgrp_opt_t *options)
 
     /* Setup a custom fapl for file accesses */
     if (custom_vol || custom_vfd) {
-        if ((tmp_fapl_id = h5tools_get_fapl(options->fapl_id, custom_vol ? &vol_info : NULL,
```

Comment:
I can't comment on the `h5tools_fopen` call below, but another case where `(params_g.fapl_id != H5P_DEFAULT) ` should probably be switched to `custom_vol || custom_vfd`